dspace-api/src/main/java/org/dspace/app/harvest/HarvestScriptConfiguration.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.harvest;
+
+import java.sql.SQLException;
+
+import org.apache.commons.cli.Options;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.scripts.configuration.ScriptConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+public class HarvestScriptConfiguration<T extends Harvest> extends ScriptConfiguration<T> {
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    private Class<T> dspaceRunnableClass;
+
+    @Override
+    public Class<T> getDspaceRunnableClass() {
+        return dspaceRunnableClass;
+    }
+
+    @Override
+    public void setDspaceRunnableClass(Class<T> dspaceRunnableClass) {
+        this.dspaceRunnableClass = dspaceRunnableClass;
+    }
+
+    public boolean isAllowedToExecute(final Context context) {
+        try {
+            return authorizeService.isAdmin(context);
+        } catch (SQLException e) {
+            throw new RuntimeException("SQLException occurred when checking if the current user is an admin", e);
+        }
+    }
+
+    public Options getOptions() {
+        Options options = new Options();
+        options.addOption("p", "purge", false, "delete all items in the collection");
+        options.addOption("r", "run", false, "run the standard harvest procedure");
+        options.addOption("g", "ping", false, "test the OAI server and set");
+        options.addOption("s", "setup", false, "Set the collection up for harvesting");
+        options.addOption("S", "start", false, "start the harvest loop");
+        options.addOption("R", "reset", false, "reset harvest status on all collections");
+        options.addOption("P", "purge", false, "purge all harvestable collections");
+        options.addOption("o", "reimport", false, "reimport all items in the collection");
```

Comment:
We should add a note to the description here that `reimport` seems to be the equivalent of `-p -r` (if I'm reading the code correctly). So, this should warn that we are purging all contents (i.e. deleting existing items) and re-importing from scratch.